tests/acceptance/10_files/can_set_sticky_bits_without_root_owning_directory.cf

```diff
@@ -0,0 +1,80 @@
+#######################################################
+#
+# Acceptance test for issue #2745
+# Ensure that cfengine creates a
+# directory not owned by root and the setgid
+# is set on the directory
+#
+#######################################################
+
+body common control
+{
+      inputs => { "../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+
+#######################################################
+
+bundle agent init
+{
+
+      #######################################
+      #Clean up test directory before testing
+      #######################################
+
+  files:
+      "$(G.testdir)/test_sgid"
+        delete => tidy;
+
+}
+
+#######################################################
+
+bundle agent test
+{
+  meta:
+
+      "description"
+        string => "Test that the agent can set the sticky
+                   bit on a directory that root does not
+                   own.";
+
+      "test_skip_unsupported"
+        string => "!windows";
+
+  vars:
+    !windows::
+      #"test_dir" string => "/tmp/test_setgid";
+      "mode" int => "2700";
+      "owner" string => "daemon";
+      "group" string => "daemon";
+
+  files:
+    !windows::
+      "$(G.testdir)/test_sgid/."
+      create => "true",
+      perms => mog( $(mode), $(owner), $(group) ),
+      classes => scoped_classes_generic("namespace", "test_sgid_dir");
+}
+
+#######################################################
+
+bundle agent check
+{
+
+  vars:
+    test_sgid_dir_reached::
+      "expect_permoct" string => "$(test.mode)";
+      "result_permoct" string => execresult('/usr/bin/stat -c %a $(G.testdir)/test_sgid', "useshell");
```

Comment:
I'm pretty sure `stat` doesn't exist on all platforms. This probably should either be changed, or we skip it on non-Linux.